### PR TITLE
Fixes the issue #382: The 'default' can't be used in 'oneOf'

### PR DESCRIPTION
**Pull request by dab-libs**

### CODE REVIEW

File: core/src/test/java/org/everit/json/schema/DefaultInOneOfTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        JSONObject jsonSchema = LOADER.readObj("default-in-last-oneof-schema.json");
+        JSONObject jsonSpecification = LOADER.readObj("default-in-oneof.json");
+        Schema schema = SchemaLoader
+                .builder()
+                .useDefaults(true)
+                .schemaJson(jsonSchema)
+                .build()
+                .load()
+                .build();
+        assertDoesNotThrow(() -> schema.validate(jsonSpecification));
```
Comment (erosb) [inline core/src/test/java/org/everit/json/schema/DefaultInOneOfTest.java:44]: Can you please verify in these 2 tests that the correct default value is picked up?

File: core/src/main/java/org/everit/json/schema/ValidationFailureReporter.java [hunk 2]
```diff
@@ -37,4 +37,12 @@ ValidationException inContextOfSchema(Schema schema, Runnable task) {
     }
 
     abstract void validationFinished();
+
+    public Object getState() {
+        return null;
+    }
+
+    public boolean isChanged(Object oldState) {
```
Comment (erosb) [inline core/src/main/java/org/everit/json/schema/ValidationFailureReporter.java:45]: Please let's keep these new methods (`#getState()` and `#isChanged()`) package-private. These are implementation-specific details and not public APIs.

**Review by erosb — CHANGES_REQUESTED** [review]
Hello @dab-libs , thank you for raising this PR. I added some comments.

File: core/src/test/java/org/everit/json/schema/DefaultInOneOfTest.java (hunk 1 above)
Comment (dab-libs) [reply core/src/test/java/org/everit/json/schema/DefaultInOneOfTest.java:47]: Done

File: core/src/main/java/org/everit/json/schema/ValidationFailureReporter.java (hunk 2 above)
Comment (dab-libs) [reply core/src/main/java/org/everit/json/schema/ValidationFailureReporter.java]: Done

**Review by erosb — APPROVED** [review]

**Comment by erosb** [comment]
Hello @dab-libs , thank you for addressing my comments, I merged the PR. Have a nice day!

**Pull request merged**